<!DOCTYPE html>
<head>
    <link rel="SHORTCUT ICON" href="./logo.ico"/>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>TSVdb</title>
    <meta name="author" content="Sun Wenjie, Ye Panmeng" />
    <meta name="description" content="TCGA splicing variants database." />
    <meta name="keywords"  content="TCGA,tumor,splicing variants ,visualization" />
    <meta name="Resource-type" content="Document" />
    <link rel="stylesheet" type="text/css" href="./css/jquery.fullPage.css" />
    <link rel="stylesheet" type="text/css" href="./css/index.css"/>
    <link rel="stylesheet" type="text/css" href="./fonts/iconfont.css">
    <!-- Global Site Tag (gtag.js) - Google Analytics -->
    <!-- <script async src="https://www.googletagmanager.com/gtag/js?id=UA-106948170-1"></script> -->
    <script>
        window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments)};
        gtag('js', new Date());

        gtag('config', 'UA-106948170-1');
    </script>
    <script type="text/javascript" src="./js/sv/check_browser.js"></script>
    <script src="./js/jquery-3.5.0.min.js"></script>
    <script src="./js/jquery-ui.min.js"></script>
    <script type="text/javascript" src="./js/scrolloverflow.min.js"></script>
    <script type="text/javascript" src="./js/jquery.fullPage.min.js"></script>
    <!--    <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=AM_HTMLorMML"></script>-->
    <script type="text/javascript">
        $(document).ready(function() {
            $('#fullpage').fullpage({
                //                sectionsColor: ['#81d4fa', '#e1f5fe', '#b3e5fc', ],
                sectionsColor: ['#e1f5fe', '#b3e5fc', '#fff'],
                navigationTooltips: ['Instruction', 'Tutorial', 'Toolchart'],
                navigation: true,
                css3: true,
                loopBottom: true,
                controlArrowColor: '#fff'
            });
        });
    </script>
</head>
<body>
    <div class="toHome">
        <a href="./index.html"><i class="iconfont icon-gene"></i>TSVdb</a>
    </div>
    <div class="toPlot">
        <a href="./plot.html">Quick Start -></a>
    </div>
    <div id="fullpage">

        <div class="section active" id="section_intro">
            <h1 class="intr_h1">TSVdb</h1>
            <h2 class="intr_h2">TCGA Splicing Variants DB</h2>
            <div class="intro"><p>TSVdb is an easy-to-use web tool for integrating and visualizing mRNA alternative splicing, transcriptional isoform expression and clinical information from The Cancer Genome Atlas project (TCGA) RNA-Seq data.</p></div>
            <!-- <div class="selection" id="selection_start"><a href="./plot_demo.html">Start</a></div> -->
            <!-- <div class="selection" id="selection_tutorial">Tutorial</div> -->
            <!--            <img src = "./image/strategy.svg" alt="strategy" style="height: 250px;" class="myImg">-->
            <img src = "./image/amination.gif" alt="strategy" style="height: 300px;" class="gif">
        </div>


        <div class="section" id="section_tutorial">
            <div class="slide" id="selecting_data">
                <h1 class="spe_h1">Tutorial</h1>
                <h2 class="sep_h2">1. Selecting the data</h2>
                <div class="slide_content">
                    <img src="./image/select_data.png" alt="select_data" style="height: 300px">
                    <p style="height: 0px"></p>
                    <p>
                    Upon opening the data visualization page, the user needs to sequentially choose the tumor type, gene, data type and clinical information. No other operations can be performed before choosing these options. Do not worry when making selections at this stage, as the options can be edited afterwards.
                    </p>
                    <p>
                    The first pop-up dialog (Fig. A) is the cancer type; please select the cancer type you are interested in and click the selection button. In the second pop-up dialog (Fig. B), please enter an Official Gene Symbol or Entrez ID. If the query does not exist, the alarm will appear and notify you. Then, choose the exon/junction usage value (Fig. C). The exon/junction usage is the scaled expression; for more information about the scaling method, please refer to the next section. Finally, choose the phenotype you are interested in and the gene expression data to sort the samples in each phenotype subgroup (Fig. D). The gene expression value can be set to another gene so that the correlation between alternative splicing for one gene and the expression of another gene can be investigated.
                    </p>
                </div>
            </div>

            <div class="slide" id="exon_junc">
                <h1 class="spe_h1">Tutorial</h1>
                <h2 class="sep_h2">2. About the graph</h2>
                <div class="slide_content">
                    <img src="./image/about_graph.png" alt="about_graph" style="height: 400px">
                    <p>
                    The options are in the header display; they can be edited by clicking the corresponding buttons in the header.
                    </p>
                    <p>
                    The samples are sorted by clinical type and then sorted by gene expression. When the mouse hovers over the clinical attribute row, the group name and a red line will appear.
                    </p>
                    <p>
                    The “Exon or Junction” gray boxes are the exons/junctions that TCGA uses to count the mapping reads in the TCGA Level 3 mRNA Seq datasets.
                    </p>
                    <p>
                    Each exon/junction usage value is shown as a shadowed line graph with the usage in the row and the samples in the columns. The exon/junction pattern will be transiently or permanently highlighted when the mouse hovers or clicks an exon/junction-shadowed line graph. Moreover, when you double click on an exon/junction-shadowed line, the UCSC Genome Browser will open in a new tab/window with the selected exon/junction area highlighted and let you examine the annotation in the UCSC Genome Browser.
                    </p>
                </div>
            </div>



            <div class="slide" id="isoform">
                <h1 class="spe_h1">Tutorial</h1>
                <h2 class="sep_h2">3. Isoform expression</h2>
                <div class="slide_content">
                    <img src="./image/about_isoform.png" alt="about_isoform" style="height: 300px">
                    <p>
                    A box plot for isoform expression (normalized RSEM) will appear after clicking the transcriptional isoform graph. On the top right side, you can see the UCSC isoform ID. Clicking on this ID will open a new tab or window display for the highlighted isoform in the UCSC Genome Browser. The buttons at the top of the pop-up box are used to choose the box plot or KM survival plot to be displayed.
                    </p>
                    <p>
                    The KM survival curve has four parts. (1) In the bottom right, there is a way to adjust the cut-off for grouping; the knob can be adjusted to change the cut-off. The default value is the middle of the range. (2) The top right displays the information on the groups, including the cut-off value and sample size of each group. (3). The top left shows the survival line for each individual included after filtering from the survival start time point. 
                    </p>
                </div>
            </div>

            <div class="slide" id="data_table">
                <h1 class="spe_h1">Tutorial</h1>
                <h2 class="sep_h2">4. Download data table</h2>
                <div class="slide_content">
                    <table class="table", style="border-style: solid">
                        <thead>
                            <tr>
                                <th style="text-align: left">Attribute</th>
                                <th>Example</th>
                                <th>Description</th>
                            </tr>
                        </thead>
                        <tbody style="text-align: left; border-style: solid">
                            <tr>
                                <td style="text-align: left">sampleID</td>
                                <td>TCGA-3L-AA1B-01A</td>
                                <td>TCGA sample id.</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">clinical_[clinical type]</td>
                                <td>Primary Solid Tumor</td>
                                <td>The phenotype information about the sample.</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">os_time</td>
                                <td>348</td>
                                <td>Overall survival time (day).</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">os_event</td>
                                <td>0</td>
                                <td>Overall survival event (0=live/censored, 1=death).</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">gene_[gene name]</td>
                                <td>100</td>
                                <td>Gene expression normalized read count.</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">isoform_[isoform id]</td>
                                <td>100</td>
                                <td>Isoform expression normalized read count.</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">exon_[location]</td>
                                <td>0.1102</td>
                                <td>Exon/junction usage value. Exon/junction usage is calculated by dividing exon/junction quantification value</br>
                                    to mean exon/junction quantification value of that person for a specific gene</td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </div>
        </div>

        <div class="section" id="toolchart">
            <h1 class="spe_h1">Tool-chains and algorithms</h1>
            <div class="slide_content">
                <img src="./image/flowchart.png" alt="toolchart" style="height: 450px; float: left; margin: 0 30px 0 0">
                <div style="height: 50px"></div>
                <p>
                The TCGA level 3 gene, isoform, exon and junction normalized read count data (UNC illuminaHiSeq_RNASeqV2) and clinical data (Merged_clinical_level_1) were downloaded from the TCGA ftp site Firehose.
                </p>
                <p>
                R was used to generate the transcriptional patterns and to perform the TCGA data cleanup. The formatted JSON data were import into MongoDB.
                </p>
                <p>
                The exon/junction usage values were derived from exon/junction quantification. First, each sample's exon/junction quantification value was divided by the expression of the gene to which it belongs. Thus, the gene expression effect was removed and the AS event was highlighted. Next, the d3.js linear scale was used to map the normalized exon/junction value to the graph coordination. The exon/junction usage value interval (domain argument in the scale function) was set to 0 - Q95 (95\% quantile) to diminish the impact of outliers. If the value of Q95 was less than 0.05, the upper bound of the interval would be set to 0.05.
            </div>
        </div>
    </div>
</body>

